capstone-backend/src/main/java/com/google/sps/servlets/ClubSearchServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.Query;
+import com.google.gson.Gson;
+import com.google.sps.data.Club;
+
+@WebServlet("/api/clubSearch")
+public class ClubSearchServlet extends HttpServlet {
+  private Firestore db;
+  private CollectionReference clubs;
+  private Gson gson;
+
+  public ClubSearchServlet() throws IOException {
+    db = Utility.getFirestoreDb();
+    clubs = db.collection("clubs");
+    gson = new Gson();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (request.getParameter("searchTerm") == null || request.getParameterMap().size() > 1) {
+      System.err.println("Error: Improper parameters, must only send searchTerm");
```

Comment:
It might be useful to split these out with separate error messages for more targeted debugging.

Reply:
No longer an issue once the clubSearch code was moved to the clubServlet. Steven's Utility code will take care of the error messages if there is no `searchTerm` present.

For the case when there is a non-null searchTerm and more than 1 parameter in the map, I have updated the error message.